linavigation-base/src/main/java/com/mapbox/navigation/base/trip/TripSession.kt

```diff
@@ -0,0 +1,32 @@
+package com.mapbox.navigation.base.trip
+
+import android.location.Location
+import com.mapbox.android.core.location.LocationEngine
+import com.mapbox.android.core.location.LocationEngineRequest
+import com.mapbox.navigation.base.route.Route
+
+interface TripSession {
+
+    val tripService: TripService
+    val locationEngine: LocationEngine
+    val locationEngineRequest: LocationEngineRequest
+    var route: Route?
+
+    fun startSession()
+    fun stopSession()
+
+    fun registerLocationObserver(locationObserver: LocationObserver)
+    fun unregisterLocationObserver(locationObserver: LocationObserver)
+
+    fun registerRouteProgressObserver(routeProgressObserver: RouteProgressObserver)
+    fun unregisterRouteProgressObserver(routeProgressObserver: RouteProgressObserver)
+
+    interface LocationObserver {
+        fun onRawLocationChanged(rawLocation: Location)
+        fun onEnhancedLocationChanged(enhancedLocation: Location)
+    }
+
+    interface RouteProgressObserver {
+        fun onRouteProgressChanged(routeProgress: RouteProgress)
```

Comment:
we'd have to add `onOffRoute` callback as well. Do you envision it as part of the `RouteProgressObserver` or separate interface? My thinking was that we want user to implement this callback by default, so it would be logical to have it as part of `RouteProgressObserver`. thoughts? also if we decide to slap off route here we should probably think of more generic name for this interface.

Reply:
> Do you envision it as part of the RouteProgressObserver or separate interface?

I don't think it's going to make much difference whether it's embedded here or not. I can introduce it now and we can iterate later if needed.

Reply:
> also if we decide to slap off route here we should probably think of more generic name for this interface.

What do you exactly have in mind here?

Reply:
Briefly looked into adding off-route events now, but it'd bring other new variables like route state. I'd lean towards tackling that together with the whole `RouteProgress` impl.

Reply:
> What do you exactly have in mind here?

Let's defer this discussion until we consider adding offroute handler